source/MRViewer/MRObjectMeshHistory.cpp

```diff
@@ -0,0 +1,31 @@
+#include "MRObjectMeshHistory.h"
+#include "MRAppendHistory.h"
+#include <MRMesh/MRObjectMesh.h>
+#include <MRMesh/MRChangeSelectionAction.h>
+#include <MRMesh/MRMesh.h>
+#include <MRMesh/MRTimer.h>
+
+namespace MR
+{
+
+void clearObjectMeshWithHistory( const std::shared_ptr<ObjectMesh>& objMesh )
+{
+    MR_TIMER
+    if ( !objMesh || !objMesh->mesh() )
+        return;
+    const auto & topology = objMesh->mesh()->topology;
+
+    // remove deleted edges from the selection
```

Comment:
Should it have SCOPED_HISTORY?

Reply:
SCOPED_HISTORY must be in the caller, a will add a comment